android/src/main/java/fr/bamlab/rnimageresizer/ImageResizer.java

```diff
@@ -207,16 +208,41 @@ private static Bitmap loadBitmap(Context context, String imagePath, BitmapFactor
     public static String createResizedImage(Context context, String imagePath, int newWidth,
                                             int newHeight, Bitmap.CompressFormat compressFormat,
                                             int quality, int rotation, String outputPath) throws IOException  {
-        // Decode the image bounds to find the size of the source image.
-        BitmapFactory.Options options = new BitmapFactory.Options();
-        options.inJustDecodeBounds = true;
-        loadBitmap(context, imagePath, options);
-
-        // Set a sample size according to the image size to lower memory usage.
-        options.inSampleSize = calculateInSampleSize(options, newWidth, newHeight);
-        options.inJustDecodeBounds = false;
-        System.out.println(options.inSampleSize);
-        Bitmap sourceImage = loadBitmap(context, imagePath, options);
+        Bitmap sourceImage = null;
+        final String base64Prefix = "data:image/";
+
+        if (imagePath.indexOf(base64Prefix) < 0) {
+            // Decode the image bounds to find the size of the source image.
+            BitmapFactory.Options options = new BitmapFactory.Options();
+            options.inJustDecodeBounds = true;
+            loadBitmap(context, imagePath, options);
+
+            // Set a sample size according to the image size to lower memory usage.
+            options.inSampleSize = calculateInSampleSize(options, newWidth, newHeight);
+            options.inJustDecodeBounds = false;
+            System.out.println(options.inSampleSize);
+            sourceImage = loadBitmap(context, imagePath, options);
+        }
+        else {
+            // base64 image.  Convert to a bitmap.
+            final int prefixLen = base64Prefix.length();
+            final boolean isJpeg = (imagePath.indexOf("jpeg") == prefixLen);
+            final boolean isPng = (!isJpeg) && (imagePath.indexOf("png") == prefixLen);
+            int commaLocation = -1;
+            if (isJpeg || isPng){
+                commaLocation = imagePath.indexOf(',');
+            }
+            if (commaLocation > 0) {
+                //Todo, can this handle both jpeg AND png?
```

Comment:
@cfogelklou Do you think you could make tests so that we can remove this TODO? :)
Also I think it would be nice to handle the loading of a base64 image into a separate function, so that we can keep createResizedImage relatively straightforward 😃